go-controller/hack/test-go.sh

```diff
@@ -22,26 +22,26 @@ function testrun {
     local otherargs="${@:3} "
     local args="-mod vendor"
     local ginkgoargs=
-
+    # enable go race detector
+    if [ ! -z "${RACE:-}" ]; then
+        args="-race "
+    fi
     if [[ "$USER" != root && " ${root_pkgs[@]} " =~ " $pkg " ]]; then
         testfile=$(mktemp --tmpdir ovn-test.XXXXXXXX)
         echo "sudo required for ${pkg}, compiling test to ${testfile}"
-        go test -covermode set -c "${pkg}" -o "${testfile}"
-        go_test="sudo ${testfile}"
+        if [[ ! -z "${RACE:-}" && "${pkg}" != "github.com/ovn-org/ovn-kubernetes/go-controller/hybrid-overlay/pkg/controller" ]]; then
```

Comment:
@alexanderConstantinescu you forgot the " # FIXME race detector fails with hybrid-overlay tests" comment here. I'll assume that means you are volunteering to fix that too ;)